code/modules/client/preferences2/preferences2.dm

```diff
@@ -0,0 +1,234 @@
+		qdel(read_player_data)
+
+	//general preferences
+	READPREF_INT(default_slot, PREFERENCE_TAG_DEFAULT_SLOT)
+	READPREF_INT(chat_toggles, PREFERENCE_TAG_CHAT_TOGGLES)
+	READPREF_INT(toggles, PREFERENCE_TAG_TOGGLES)
+	READPREF_INT(toggles2, PREFERENCE_TAG_TOGGLES2)
+	READPREF_INT(clientfps, PREFERENCE_TAG_CLIENTFPS)
+	READPREF_INT(parallax, PREFERENCE_TAG_PARALLAX)
+	READPREF_INT(pixel_size, PREFERENCE_TAG_PIXELSIZE)
+	READPREF_INT(tip_delay, PREFERENCE_TAG_TIP_DELAY)
+
+	READPREF_RAW(asaycolor, PREFERENCE_TAG_ASAY_COLOUR)
+	READPREF_RAW(ooccolor, PREFERENCE_TAG_OOC_COLOUR)
+	READPREF_RAW(lastchangelog, PREFERENCE_TAG_LAST_CL)
+	READPREF_RAW(UI_style, PREFERENCE_TAG_UI_STYLE)
+	READPREF_RAW(outline_color, PREFERENCE_TAG_OUTLINE_COLOUR)
+	READPREF_RAW(see_balloon_alerts, PREFERENCE_TAG_BALLOON_ALERTS)
+	READPREF_RAW(scaling_method, PREFERENCE_TAG_SCALING_METHOD)
+	READPREF_RAW(ghost_form, PREFERENCE_TAG_GHOST_FORM)
+	READPREF_RAW(ghost_orbit, PREFERENCE_TAG_GHOST_ORBIT)
+	READPREF_RAW(ghost_accs, PREFERENCE_TAG_GHOST_ACCS)
+	READPREF_RAW(ghost_others, PREFERENCE_TAG_GHOST_OTHERS)
+	READPREF_RAW(pda_style, PREFERENCE_TAG_PDA_STYLE)
+	READPREF_RAW(pda_color, PREFERENCE_TAG_PDA_COLOUR)
+
+	READPREF_JSONDEC(ignoring, PREFERENCE_TAG_IGNORING)
+	READPREF_JSONDEC(key_bindings, PREFERENCE_TAG_KEYBINDS)
+	READPREF_JSONDEC(purchased_gear, PREFERENCE_TAG_PURCHASED_GEAR)
+	READPREF_JSONDEC(be_special, PREFERENCE_TAG_BE_SPECIAL)
+
+	//Sanitize
+	asaycolor		= sanitize_ooccolor(sanitize_hexcolor(asaycolor, 6, TRUE, initial(asaycolor)))
+	ooccolor		= sanitize_ooccolor(sanitize_hexcolor(ooccolor, 6, TRUE, initial(ooccolor)))
+	lastchangelog	= sanitize_text(lastchangelog, initial(lastchangelog))
+	UI_style		= sanitize_inlist(UI_style, GLOB.available_ui_styles, GLOB.available_ui_styles[1])
+
+	default_slot	= sanitize_integer(default_slot, TRUE, max_usable_slots, initial(default_slot))
+	toggles			= sanitize_integer(toggles, FALSE, INFINITY, initial(toggles)) // yes
+	toggles2		= sanitize_integer(toggles2, FALSE, INFINITY, initial(toggles2))
+	clientfps		= sanitize_integer(clientfps, FALSE, 1000, FALSE)
+	parallax		= sanitize_integer(parallax, PARALLAX_INSANE, PARALLAX_DISABLE, null)
+
+	pixel_size		= sanitize_float(pixel_size, PIXEL_SCALING_AUTO, PIXEL_SCALING_3X, 0.5, initial(pixel_size))
+	scaling_method  = sanitize_text(scaling_method, initial(scaling_method))
+	ghost_form		= sanitize_inlist(ghost_form, GLOB.ghost_forms, initial(ghost_form))
+	ghost_orbit 	= sanitize_inlist(ghost_orbit, GLOB.ghost_orbits, initial(ghost_orbit))
+	ghost_accs		= sanitize_inlist(ghost_accs, GLOB.ghost_accs_options, GHOST_ACCS_DEFAULT_OPTION)
+	ghost_others	= sanitize_inlist(ghost_others, GLOB.ghost_others_options, GHOST_OTHERS_DEFAULT_OPTION)
+	be_special		= SANITIZE_LIST(be_special)
+
+	pda_style		= sanitize_inlist(pda_style, GLOB.pda_styles, initial(pda_style))
+	pda_color		= sanitize_hexcolor(pda_color, 6, TRUE, initial(pda_color))
```

Comment:
these are not persisting for some reason

Reply:
Will look into this.

Reply:
This seems to be due to a weird conflict with TGS and not the code. Testing locally and it all works as expected. Doing a fresh close and pull also has those two options working as expected again.